### DISCUSSION TITLE
Namespace in library xml manifest

### DISCUSSION THREAD

**Discussion by sergeytolkachyov** [joomla/joomla-cms#39342] [2. Cold Features]
Now, when creating our own field types for modules, plugins, we specify the `addfieldprefix` attribute, where the namespace for the field class is specified. Usually these fields are located together with the module or plugin itself. What if the field classes are supplied with the library? The namespace element is not supposed to be specified in the xml manifest of the library. Thus, it is not possible to use this namespace in the `addfieldprefix `attribute.

**Comment by sergeytolkachyov** [comment]
I found that is possible yet, but there is no documentation

**Comment by sergeytolkachyov** [comment]
Do we still need in additional plugin for register library namespace like here described? https://docs.joomla.org/Using_own_library_in_your_extensions#Namespace_Loader I think no.

**Reply by dgrammatiko** [reply]
> Do we still need in additional plugin for register library namespace like here described?

Libraries with correct namespace are automatically loaded so you don't need any plugin. Check how it can be correctly done: https://github.com/ttc-freebies/plugin-responsive-images/blob/4.0.0/src/libraries/Ttc/Ttc.xml

**Comment by sergeytolkachyov** [comment]
@dgrammatiko thanks, I found this. But if my library has name like Vendor/Library and file structure is Vendor/Library/src instead Library/src can I specify <namespace path="Library/src">Vendor\Library</namespace> ?

**Reply by dgrammatiko** [reply]
No you CANOT use the `vendor` folder as it belongs to Joomla. You could have a `libraries/YourLibrary/src/vendor` if that's working for you but you SHOULD NOT touch the `libraries/vendor` folder

**Reply by sergeytolkachyov** [reply]
I was on the phone and expressed myself incorrectly. In Joomla 3 I installed all my libraries in my own directory in `libraries` folder. 
![image](https://user-images.githubusercontent.com/6236403/205286296-37ec6ece-fb58-409f-bdc0-ccccbe4dc52e.png)
![image](https://user-images.githubusercontent.com/6236403/205286474-c73ceda3-56ec-40d8-b784-6eb51f633d3a.png)
![image](https://user-images.githubusercontent.com/6236403/205286597-3aae28d0-d616-4b2a-ab5d-2b63540f0239.png)
All these libraries are independent packages and I want to install its to my "vendor" - "Webtolk" - folder. So the path will be `libraries/Webtolk/Amocrm/src` or `libraries/Webtolk/Pochtaru/src`
Like this
![image](https://user-images.githubusercontent.com/6236403/205287846-cc0beaf2-f3d4-477e-b703-ee64f8225e42.png)
So, can I specify `<namespace path="Webtolk/Amocrm/src">Webtolk/Amocrm</namespace>`, `<namespace path="Webtolk/Otpravkapochtaru/src">Webtolk/Otpravkapochtaru</namespace>`

OR
it could be like this
![image](https://user-images.githubusercontent.com/6236403/205288453-57457c18-3b32-4c4c-8208-cafb2a924c68.png)
and in xml manifest for each library I will specify `<libraryname>Webtolk/src/Amocrm</libraryname>` (OMG! But there are independent libraries. How to separate?) What will I specify in <namespace path="??????">Webtolk\Amocrm</namespace>

I'm confused)))

**Reply by sergeytolkachyov** [reply]
I understand the way for single library per folder, but for several inpenendent libraries - i don't understand
